codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/error/OperationErrorGenerator.kt

```diff
@@ -103,12 +85,25 @@ class OperationErrorGenerator(
             rust(
                 """
                 /// An unexpected error occurred (e.g., invalid JSON returned by the service or an unknown error code).
+                ##[deprecated(note = "Don't directly match on `Unhandled`. Instead, match on `_` and use \
```

Comment:
should probably suggest `err` again—if you match on `_` and reach back up to the error, sometimes you'll hit lifetime issues